Test_byteToTime and Test_scmDataToTime failures on 32bit architectures

When building on i686 and armv7hl:

```
Testing    in: /builddir/build/BUILD/ptp-5062d2bca401ff0fc1f40d10060b68004cac64df/_build/src
         PATH: /builddir/build/BUILD/ptp-5062d2bca401ff0fc1f40d10060b68004cac64df/_build/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/sbin
       GOPATH: /builddir/build/BUILD/ptp-5062d2bca401ff0fc1f40d10060b68004cac64df/_build:/usr/share/gocode
  GO111MODULE: off
      command: go test -buildmode pie -compiler gc -ldflags " -X github.com/facebookincubator/ptp/version=0 -X github.com/facebookincubator/ptp/version.commit=5062d2bca401ff0fc1f40d10060b68004cac64df -extldflags '-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld  '"
      testing: github.com/facebookincubator/ptp
github.com/facebookincubator/ptp/protocol
--- FAIL: Test_byteToTime (0.00s)
    timestamp_test.go:34: 
        	Error Trace:	timestamp_test.go:34
        	Error:      	Not equal: 
        	            	expected: 1612028735717200436
        	            	actual  : 1612028735000000000
        	Test:       	Test_byteToTime
--- FAIL: Test_scmDataToTime (0.00s)
    --- FAIL: Test_scmDataToTime/hardware_timestamp (0.00s)
        timestamp_test.go:106: 
            	Error Trace:	timestamp_test.go:106
            	Error:      	Not equal: 
            	            	expected: 1612028735717200436
            	            	actual  : 1612028735000000000
            	Test:       	Test_scmDataToTime/hardware_timestamp
    --- FAIL: Test_scmDataToTime/software_timestamp (0.00s)
        timestamp_test.go:106: 
            	Error Trace:	timestamp_test.go:106
            	Error:      	Not equal: 
            	            	expected: 1612028735717200436
            	            	actual  : 1612028735000000000
            	Test:       	Test_scmDataToTime/software_timestamp
FAIL
exit status 1
```
See https://koji.fedoraproject.org/koji/taskinfo?taskID=70310854 and https://koji.fedoraproject.org/koji/taskinfo?taskID=70310855 for the full log (look at build.log).

--- a/protocol/timestamp.go
+++ b/protocol/timestamp.go
@@ -160,8 +160,10 @@ feature. Only one field is non-zero at any time. Most timestamps
 are passed in ts[0]. Hardware timestamps are passed in ts[2].
 */
 func scmDataToTime(data []byte) (ts time.Time, err error) {
+	// on 32bit platforms Timespec is 8bits, while on 64bit it's 16bits
+	size := unsafe.Sizeof(unix.Timespec{})
 	// first, try to use hardware timestamps
-	ts, err = byteToTime(data[32:48])
+	ts, err = byteToTime(data[size*2 : size*3])
 	if err != nil {
 		return ts, err
 	}
@@ -169,7 +171,7 @@ func scmDataToTime(data []byte) (ts time.Time, err error) {
 	// we can't use ts.IsZero because for some crazy reason timestamp parsed using time.Unix()
 	// reports IsZero() == false, even if seconds and nanoseconds are zero.
 	if ts.UnixNano() == 0 {
-		ts, err = byteToTime(data[0:16])
+		ts, err = byteToTime(data[0:size])
 		if err != nil {
 			return ts, err
 		}

--- a/protocol/timestamp_test.go
+++ b/protocol/timestamp_test.go
@@ -19,6 +19,7 @@ package protocol
 import (
 	"fmt"
 	"net"
+	"strconv"
 	"testing"
 	"time"
 
@@ -28,6 +29,10 @@ import (
 // Testing conversion so if Packet structure changes we notice
 func Test_byteToTime(t *testing.T) {
 	timeb := []byte{63, 155, 21, 96, 0, 0, 0, 0, 52, 156, 191, 42, 0, 0, 0, 0}
+	// time_t is 32 bit on Linux 386
+	if strconv.IntSize == 32 {
+		timeb = []byte{63, 155, 21, 96, 52, 156, 191, 42}
+	}
 	res, err := byteToTime(timeb)
 	require.Nil(t, err)
 
@@ -59,6 +64,39 @@ func Test_ReadTXtimestamp(t *testing.T) {
 }
 
 func Test_scmDataToTime(t *testing.T) {
+	hwData := []byte{
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		63, 155, 21, 96, 0, 0, 0, 0, 52, 156, 191, 42, 0, 0, 0, 0,
+	}
+	swData := []byte{
+		63, 155, 21, 96, 0, 0, 0, 0, 52, 156, 191, 42, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	}
+	noData := []byte{
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+		0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+	}
+	if strconv.IntSize == 32 {
+		hwData = []byte{
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			63, 155, 21, 96, 52, 156, 191, 42,
+		}
+		swData = []byte{
+			63, 155, 21, 96, 52, 156, 191, 42,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+		}
+		noData = []byte{
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+			0, 0, 0, 0, 0, 0, 0, 0,
+		}
+	}
+
 	tests := []struct {
 		name    string
 		data    []byte
@@ -66,32 +104,20 @@ func Test_scmDataToTime(t *testing.T) {
 		wantErr bool
 	}{
 		{
-			name: "hardware timestamp",
-			data: []byte{
-				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-				63, 155, 21, 96, 0, 0, 0, 0, 52, 156, 191, 42, 0, 0, 0, 0,
-			},
+			name:    "hardware timestamp",
+			data:    hwData,
 			want:    1612028735717200436,
 			wantErr: false,
 		},
 		{
-			name: "software timestamp",
-			data: []byte{
-				63, 155, 21, 96, 0, 0, 0, 0, 52, 156, 191, 42, 0, 0, 0, 0,
-				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-			},
+			name:    "software timestamp",
+			data:    swData,
 			want:    1612028735717200436,
 			wantErr: false,
 		},
 		{
-			name: "zero timestamp",
-			data: []byte{
-				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-				0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-			},
+			name:    "zero timestamp",
+			data:    noData,
 			want:    0,
 			wantErr: true,
 		},

